### ISSUE TITLE
RFC: Multiple Search Dropdown -> not clearing search query on select

### ISSUE THREAD

**Issue by creedmangrum** [Semantic-Org/Semantic-UI-React#1954]
This isn't a bug as the code is intentional, but I'm curious as to why in the search dropdown if multiple items are allowed, why the search query is not cleared on enter of item click? I would be happy to help out and make this a prop option, unless you would say otherwise. It seems like something useful as I can't really think of a use case for leaving the searchQuery after select, though I'm sure there are some. If this would be something that would help, happy to contribute, just want to make sure that I wouldn't be totally off base.

Thanks for all your work on this!

### Steps
search for an item on multi select, then select an item that matches

### Expected Result
i would expect the search query to be cleared

### Actual Result
search query remains. it is due to

```
    if (!multiple) this.clearSearchQuery();
```
that is in both on enter and item click.

### Version
0.71.4

**Comment by levithomason** [comment]
**tl;dr** When selecting many options with the same query required, you have to enter after every selection. Since it poses no usability issues keeping the query, we've defaulted to this option.

For more, see https://github.com/Semantic-Org/Semantic-UI-React/pull/1795#discussion_r128802050

>### handleItemClick
>
>This currently resets the query when an menu item is made active, however, I have always found this to be a usability issue with `multiple` dropdowns.  When there are lots of items, it is often more helpful to keep the query while clicking items.  Otherwise, the user has to enter the same query over and over for every item.  Example, selecting all states that start with `ma`:
>
>![http://g.recordit.co/OqqnZDHR5u.gif](http://g.recordit.co/OqqnZDHR5u.gif)

Will move the discussion to the PR.

**Comment by AndrewCi** [comment]
I believe there was a regression somehow and this issue should be re-opened #4063 @levithomason @creedmangrum
